packages/babel-plugin-transform-es2015-for-of/src/index.js

```diff
@@ -92,7 +92,7 @@ export default function ({ messages, template, types: t }) {
   return {
     visitor: {
       ForOfStatement(path, state) {
-        if (path.get("right").isArrayExpression()) {
+        if (path.get("right").isArrayExpression() || path.get("right").isGenericType("Array")) {
```

Comment:
@hzoo has this optimization landed in master yet? It would be amazing to have this land since it'll allow us to make optimizations to our code without using loose mode